File: multi-storage-transaction-sample/src/main/java/sample/command/GetOrdersCommand.java
package sample.command;

import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.Sample;

@Command(name = "GetOrders", description = "Get order information by customer ID")
public class GetOrdersCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "CUSTOMER_ID", description = "customer ID")
  private int customerId;

  @Override
  public Integer call() throws Exception {
    try (Sample sample = new Sample()) {
      System.out.println(sample.getOrdersByCustomerId(customerId));
    }
    return 0;
  }
}

File: scalardb-dotnet-samples/scalardb-cluster-sample/Commands/GetOrdersCommand.cs
using System.CommandLine;

namespace ScalarDbClusterSample.Commands;

public static class GetOrdersCommand
{
    private const string Name = "GetOrders";
    private const string Description = "Get information about orders by customer ID";

    private const string ArgName = "customer_id";
    private const string ArgDescription = "customer ID";

    public static Command Create()
    {
        var customerIdArg = new Argument<int>(ArgName, ArgDescription);
        var getOrdersCommand = new Command(Name, Description)
        {
            customerIdArg
        };

        getOrdersCommand.SetHandler(async customerId =>
        {
            using var sample = new Sample();
            var orders = await sample.GetOrdersByCustomerId(customerId);
            
            Console.WriteLine(orders);
        }, customerIdArg);
        
        return getOrdersCommand;
    }
}
